tests/components/ring/test_init.py

```diff
@@ -266,3 +270,109 @@ async def test_issue_deprecated_service_ring_update(
         "This is deprecated and will stop working in Home Assistant 2024.10. "
         "Use 'homeassistant.update_entity' instead which updates all ring entities"
     ) in caplog.text
+
+
+@pytest.mark.parametrize(
+    ("domain", "old_unique_id"),
+    [
+        (
+            LIGHT_DOMAIN,
+            123456,
+        ),
+        (
+            CAMERA_DOMAIN,
+            654321,
+        ),
+        (
+            CAMERA_DOMAIN,
+            "99999",
+        ),
+    ],
+)
+async def test_update_unique_id(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    caplog: pytest.LogCaptureFixture,
+    requests_mock: requests_mock.Mocker,
+    domain: str,
+    old_unique_id: int | str,
+) -> None:
+    """Test unique_id update of integration."""
+    entry = MockConfigEntry(
+        title="Ring",
+        domain=DOMAIN,
+        data={
+            CONF_USERNAME: "foo@bar.com",
+            "token": {"access_token": "mock-token"},
+        },
+        unique_id="foo@bar.com",
+    )
+    entry.add_to_hass(hass)
+
+    entity: er.RegistryEntry = entity_registry.async_get_or_create(
+        domain=domain,
+        platform=DOMAIN,
+        unique_id=old_unique_id,
+        config_entry=entry,
+    )
+    assert entity.unique_id == old_unique_id
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity_migrated = entity_registry.async_get(entity.entity_id)
+    assert entity_migrated
+    assert entity_migrated.unique_id == str(old_unique_id)
+    assert (
+        (f"Fixing non string unique id {old_unique_id}") in caplog.text
+    ) == isinstance(old_unique_id, int)
```

Comment:
This looks a bit off

Reply:
Removed the string unique id into a separate test.